api/controllers/task.go

```diff
@@ -0,0 +1,84 @@
+package controllers
+
+import (
+	"encoding/json"
+
+	"github.com/quintilesims/layer0/api/provider"
+	"github.com/quintilesims/layer0/api/scheduler"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/zpatrick/fireball"
+)
+
+type TaskController struct {
+	TaskProvider provider.TaskProvider
+	JobScheduler scheduler.JobScheduler
+}
+
+func NewTaskController(t provider.TaskProvider) *TaskController {
+	return &TaskController{
+		TaskProvider: t,
+	}
+}
+
+func (t *TaskController) Routes() []*fireball.Route {
+	return []*fireball.Route{
+		{
+			Path: "/task",
+			Handlers: fireball.Handlers{
+				"GET":  t.ListTasks,
+				"POST": t.CreateTask,
+			},
+		},
+		{
+			Path: "/task/:id",
+			Handlers: fireball.Handlers{
+				"GET":    t.GetTask,
+				"DELETE": t.DeleteTask,
+			},
+		},
+	}
+}
+
+func (t *TaskController) CreateTask(c *fireball.Context) (fireball.Response, error) {
+	var req models.CreateTaskRequest
+	if err := json.NewDecoder(c.Request.Body).Decode(&req); err != nil {
+		return nil, errors.New(errors.InvalidRequest, err)
+	}
+
+	model, err := t.TaskProvider.Create(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return fireball.NewJSONResponse(202, model)
+}
+
+func (t *TaskController) DeleteTask(c *fireball.Context) (fireball.Response, error) {
+	id := c.PathVariables["id"]
+	if err := t.TaskProvider.Delete(id); err != nil {
```

Comment:
I think we're still a unclear on how tasks will look. This may look radically different later on